codecs/h264_packet_test.go

```diff
@@ -291,3 +292,35 @@ func TestH264Payloader_Payload_SPS_and_PPS_handling(t *testing.T) {
 		t.Fatal("SPS and PPS aren't packed together")
 	}
 }
+
+func TestH264Payloader_PayloadAndDetectDataIframeAfterSpsPps(t *testing.T) {
+	pck := H264Payloader{}
+
+	// When packetizing SPS and PPS are emitted with following NALU
+	res := pck.Payload(1500, []byte{0x07, 0x00, 0x01})
```

Comment:
You did change the h264 payloader to call PayloadAndDetectData so why don't you use it for SPS/PPS as well? (And at the same time should assert both len(res) == 0 and idx == -1

Reply:
ok.. good idea